### DISCUSSION TITLE
关于DDL上线目的详情描述

### DISCUSSION THREAD

**Discussion by czxin788** [hhyo/Archery#1890] [Ideas]
大数据工程师有一个痛点，就是想知道每个DDL语句是哪个项目的，上线目的是什么。
于是，我对archery做了这样的调整，只针对DDL语句（我已经做了DDL和DML分离）：
![image](https://user-images.githubusercontent.com/13861904/192926527-871d5cec-7734-4c17-94b6-29e2303ab5e5.png)

![image](https://user-images.githubusercontent.com/13861904/192925964-374315e9-ad01-443c-9a93-419e41286ece.png)

![image](https://user-images.githubusercontent.com/13861904/192926176-53df374d-1e3e-43ce-bf01-356464e62940.png)

**Comment by boldmanQ** [comment]
开发人员在上线工单的详情页选择DML还是DDL，如果是DDL，前端页面则增加描述框？

**Reply by czxin788** [reply]
对，
选择DDL会自动弹出描述框；
选择DML，会自动收回描述框。

**Reply by boldmanQ** [reply]
现在Archery的审核流程是不区分DML和DDL的
我们大数据的人员也需要知悉并审核DDL，请问你Github主页的[Archery1](https://github.com/czxin788/Archery1)项目包含你增加功能的代码嘛？

**Reply by czxin788** [reply]
不包含，那是全新的archery 1.9.1版本，我只改了其中的mongo.py文件，然后做的PR。

**Comment by LeoQuote** [comment]
目的这块是用需求链接完成的

**Reply by boldmanQ** [reply]
内部有讨论过根据工单类型(DDL/DML)设置不同的审批流程吗?
[discussions: 1688](https://github.com/hhyo/Archery/discussions/1688#discussioncomment-3171306)

**Reply by czxin788** [reply]
DDL和DML这块我用前端js控制的，
1、当以create、alter、drop开头的语句，就判定为DDL，自动匹配流程到：开发负责人-->DBA；
2、当以insert、update、delete开头的语句，就判定为dml语句，自动匹配流程到：开发负责人并执行；
3、数据层的功能只由开发负责人进行审批，表结构变更的工单由开发负责人和DBA共同审批；
4、用户选择了DDL语句，却写了dml语句，会弹窗告警：
![image](https://user-images.githubusercontent.com/13861904/210310951-dec3efde-1c46-46eb-a369-29752ca715d1.png)
